src/software/gui/standalone_simulator/widgets/standalone_simulator_draw_function_visualizer.cpp

```diff
@@ -21,21 +24,46 @@ void StandaloneSimulatorDrawFunctionVisualizer::setStandaloneSimulator(
     standalone_simulator = simulator;
 }
 
+void StandaloneSimulatorDrawFunctionVisualizer::keyPressEvent(QKeyEvent* event)
+{
+    if (event->key() == Qt::Key_R)
+    {
+        r_clicked = true;
+    }
+}
+
+void StandaloneSimulatorDrawFunctionVisualizer::resetFlags()
+{
+    initial_click_point = Point(0, 0);
+    final_click_point   = Point(0, 0);
+    ctrl_clicked        = false;
+    shift_clicked       = false;
+    r_clicked           = false;
+}
+
 void StandaloneSimulatorDrawFunctionVisualizer::mousePressEvent(QMouseEvent* event)
 {
     // If Ctrl is pressed, place the ball where the user clicks
     if (event->modifiers() & Qt::ControlModifier && standalone_simulator)
     {
-        ctrl_clicked  = true;
-        initial_point = createPoint(mapToScene(event->pos()));
-        final_point   = createPoint(mapToScene(event->pos()));
-        standalone_simulator->setBallState(BallState(initial_point, Vector(0, 0)));
+        ctrl_clicked        = true;
+        initial_click_point = createPoint(mapToScene(event->pos()));
+        final_click_point   = createPoint(mapToScene(event->pos()));
+        standalone_simulator->setBallState(BallState(initial_click_point, Vector(0, 0)));
     }
     else if (event->modifiers() & Qt::ShiftModifier && standalone_simulator)
     {
+        shift_clicked        = true;
         Point point_in_scene = createPoint(mapToScene(event->pos()));
         robot                = standalone_simulator->getRobotAtPosition(point_in_scene);
     }
+    else if (r_clicked && standalone_simulator)
+    {
+        initial_click_point = createPoint(mapToScene(event->pos()));
+        robot = standalone_simulator->getRobotAtPosition(initial_click_point);
+        auto physics_robot  = robot.lock();
+        initial_click_point = physics_robot->position();
```

Comment:
I get a segfault here when physics_robot is null. If we're going to do it like this, then we need to check if physics_robot is not null

Reply:
Yeah that was silly, I should have tested it more. I've fixed it and messed around with it trying to break it. Seems to work fine now